src/CloudFoundry.VisualStudio/MSBuild/MSBuildProcess.cs

```diff
@@ -0,0 +1,105 @@
+        {
+            System.Threading.Tasks.Task.Factory.StartNew(() =>
+            {
+                DTE dte = (DTE)CloudFoundry_VisualStudioPackage.GetGlobalService(typeof(DTE));
+                var window = dte.Windows.Item(EnvDTE.Constants.vsWindowKindOutput);
+                var output = (OutputWindow)window.Object;
+                bool paneFound = false;
+
+                OutputWindowPane pane = null;
+
+                try
+                {
+                    foreach (OutputWindowPane item in output.OutputWindowPanes)
+                    {
+                        if (item.Name == "Publish")
+                        {
+                            pane = output.OutputWindowPanes.Item("Publish");
+                            paneFound = true;
+                            break;
+                         }
+                    }
+
+                    if (paneFound == false)
+                    {
+                        pane = output.OutputWindowPanes.Add("Publish");
+                    }
+                    pane.Activate();
+                }
+                catch (Exception ex)
+                {
+                    throw ex;
+                }
+
+                
+                var engine = InitializeMsBuildEngine(null);
+
+                ErrorListProvider errorList = CloudFoundry_VisualStudioPackage.GetErrorList();
+
+                if (errorList == null)
+                {
+                    throw new InvalidOperationException("Could not retrieve error list provider");
+                }
+                
+                errorList.Tasks.Clear();
+
+                Microsoft.Build.Framework.ILogger fileLogger = new Microsoft.Build.Logging.FileLogger();
+
+                MSBuildLogger customLogger = new MSBuildLogger(pane, errorList);
+
+                customLogger.Verbosity = verbosity;
+
+                fileLogger.Parameters = string.Format("logfile={0}", logFile);
+
+                fileLogger.Verbosity = verbosity;
+
+                engine.RegisterLogger(fileLogger);
+                engine.RegisterLogger(customLogger);
+
+                engine.SetGlobalProperty("DeployOnBuild", "true");
+                engine.SetGlobalProperty("PublishProfile", pubxml);
+
+                foreach (var projectItem in engine.LoadedProjects)
+                {
+                    if (projectItem.FullPath.Equals(projectFullPath))
+                    {
+                        projectItem.Build(new string[2] { "Build", "WebCloudFoundryPublish" });
+                    }
+                }
+
+                if (errorList.Tasks.Count > 0)
+                {
+                    errorList.Show();
+                }
+
+                fileLogger.Shutdown();
+                engine.UnregisterAllLoggers();
+            });
+        }
+
+        private ProjectCollection InitializeMsBuildEngine(ProjectCollection existingEngine)
```

Comment:
This method is only called once, with `null` as a parameter. Please remove the parameter, and update the implementation.